### PR TITLE
Use TypeConverter instead of a separate function for converting types.

### PR DESCRIPTION
**Pull request by tungld**
Currently the function `convertToMemRefType` does the job of type conversion in a ConversionPattern. While every ConversionPattern has a type converter, there is no reason to use a separate function like `convertToMemRefType`.

This patch will replace `converToMemRefType` by the ConversionPattern's TypeConverter.

Signed-off-by: Tung D. Le <tung@jp.ibm.com>

### CODE REVIEW

File: src/Conversion/ONNXToKrnl/Math/Clip.cpp [hunk 1]
```diff
@@ -32,7 +32,12 @@ struct ONNXClipOpLowering : public ConversionPattern {
       ConversionPatternRewriter &rewriter) const final {
     Location loc = op->getLoc();
     ONNXClipOp clipOp = cast<ONNXClipOp>(op);
-    MemRefType memRefType = convertToMemRefType(*op->result_type_begin());
+
+    // Convert the output type to MemRefType.
+    Type convertedType = typeConverter->convertType(*op->result_type_begin());
+    if (!convertedType || !convertedType.isa<MemRefType>())
```
Comment (etiotto) [inline src/Conversion/ONNXToKrnl/Math/Clip.cpp:38]: We expect the type to be converted correctly, so should this be an assert?

Comment (tungld) [reply src/Conversion/ONNXToKrnl/Math/Clip.cpp:38]: OK. I'll change to use `assert`.

**Review by etiotto — APPROVED** [review]
LGTM.

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by tungld** [comment]
@etiotto thanks, I addressed your comments.

**Pull request merged**